### PR TITLE
Issue 1281/add cancel event

### LINKED ISSUE #1281: Cancellning events
## Description
A user should be able to cancel an event.

## Requirements
- [x] Clicking the ellipsis menu to the right of the Publish button should show a Cancel option.
- [x] Clicking Cancel should PATCH the event with `{ cancelled: currentDate }`

## Design specifications'
bild

### PR DESCRIPTION
**Pull request by rebecarubio**
## Description
This PR adds the menu options to cancel and restore an event

## Screenshots
https://user-images.githubusercontent.com/36491300/236836262-522183ac-2525-4e57-a176-ab74b6a8e436.mp4

## Changes
* Adds cancel and restore methods in event Model
* Adds cancel and restore methods in the ZUIEllipsisMenu component

## Notes to reviewer
None

## Related issues
Resolves #1281

### CODE REVIEW

File: src/features/events/components/EventActionButtons.tsx [hunk 1]
```diff
@@ -60,15 +66,37 @@ const EventActionButtons: React.FunctionComponent<EventActionButtonsProps> = ({
...
+                        eventTitle: event.title || event.activity.title,
+                      }),
+                });
+              },
+
+              startIcon: event.cancelled ? (
+                <RestoreOutlined />
+              ) : (
+                <CancelOutlined />
+              ),
```
Comment (richardolsson) [inline src/features/events/components/EventActionButtons.tsx:96]: Nice! 💯

File: src/features/events/models/EventDataModel.ts [hunk 2]
```diff
@@ -213,6 +225,8 @@ export default class EventDataModel extends ModelBase {
 
       if (startTime > now) {
         return EventState.SCHEDULED;
+      } else if (data.cancelled) {
+        return EventState.CANCELLED;
```
Comment (richardolsson) [inline src/features/events/models/EventDataModel.ts:229]: This logic is not working as I would expect. If an event is cancelled, everything else should be irrelevant, so I think this if statement needs to be at the very top, i.e. above line 222.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work! I just couldn't get the chip to update as I would expect, and I think it's because a logic error that I've pointed out below. 👇 

This is the event that doesn't update as I would expect: http://localhost:3000/organize/1/projects/standalone/events/172

File: src/features/events/models/EventDataModel.ts (hunk 2 above)
Comment (rebecarubio) [reply src/features/events/models/EventDataModel.ts:229]: Of course! I'll change it asap!

**Review by richardolsson — APPROVED** [review]
Nice! This is working as expected now. 💯

**Pull request merged**